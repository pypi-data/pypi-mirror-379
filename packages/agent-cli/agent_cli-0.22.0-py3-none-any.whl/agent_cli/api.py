"""FastAPI web service for Agent CLI transcription."""

from __future__ import annotations

import logging
from pathlib import Path
from typing import Annotated, Any

from fastapi import Depends, FastAPI, File, Form, HTTPException, Request, UploadFile
from pydantic import BaseModel

from agent_cli import config, opts
from agent_cli.agents.transcribe import AGENT_INSTRUCTIONS, INSTRUCTION, SYSTEM_PROMPT
from agent_cli.core.audio_format import VALID_EXTENSIONS, convert_audio_to_wyoming_format
from agent_cli.core.transcription_logger import get_default_logger
from agent_cli.services import asr
from agent_cli.services.llm import process_and_update_clipboard

# Configure logging
logging.basicConfig(level=logging.INFO)
LOGGER = logging.getLogger(__name__)

app = FastAPI(
    title="Agent CLI Transcription API",
    description="Web service for audio transcription and text cleanup",
    version="1.0.0",
)


@app.middleware("http")
async def log_requests(request: Request, call_next) -> Any:  # type: ignore[no-untyped-def]  # noqa: ANN001
    """Log basic request information."""
    client_ip = request.client.host if request.client else "unknown"
    LOGGER.info("%s %s from %s", request.method, request.url.path, client_ip)

    response = await call_next(request)

    if response.status_code >= 400:  # noqa: PLR2004
        LOGGER.warning(
            "Request failed: %s %s â†’ %d",
            request.method,
            request.url.path,
            response.status_code,
        )

    return response


class TranscriptionResponse(BaseModel):
    """Response model for transcription endpoint."""

    raw_transcript: str
    cleaned_transcript: str | None = None
    success: bool
    error: str | None = None


class HealthResponse(BaseModel):
    """Response model for health check."""

    status: str
    version: str


class TranscriptionRequest(BaseModel):
    """Request model for transcription endpoint."""

    cleanup: bool = True
    extra_instructions: str | None = None


async def parse_transcription_form(
    cleanup: Annotated[str | bool, Form()] = True,
    extra_instructions: Annotated[str | None, Form()] = None,
) -> TranscriptionRequest:
    """Parse form data into TranscriptionRequest model."""
    cleanup_bool = cleanup.lower() in ("true", "1", "yes") if isinstance(cleanup, str) else cleanup
    return TranscriptionRequest(cleanup=cleanup_bool, extra_instructions=extra_instructions)


@app.get("/health", response_model=HealthResponse)
async def health_check() -> HealthResponse:
    """Health check endpoint."""
    return HealthResponse(status="healthy", version="1.0.0")


async def _transcribe_with_provider(
    audio_data: bytes,
    provider_cfg: config.ProviderSelection,
    wyoming_asr_cfg: config.WyomingASR,
    openai_asr_cfg: config.OpenAIASR,
) -> str:
    """Transcribe audio using the configured provider."""
    transcriber = asr.create_recorded_audio_transcriber(provider_cfg)

    if provider_cfg.asr_provider == "local":
        return await transcriber(
            audio_data=audio_data,
            wyoming_asr_cfg=wyoming_asr_cfg,
            logger=LOGGER,
        )
    if provider_cfg.asr_provider == "openai":
        return await transcriber(
            audio_data=audio_data,
            openai_asr_cfg=openai_asr_cfg,
            logger=LOGGER,
        )
    msg = f"Unsupported ASR provider: {provider_cfg.asr_provider}"
    raise ValueError(msg)


def _is_valid_audio_file(value: Any) -> bool:
    """Check if the provided value is a valid audio file."""
    return (
        hasattr(value, "filename")
        and hasattr(value, "content_type")
        and (
            (value.content_type and value.content_type.startswith("audio/"))
            or (value.filename and value.filename.lower().endswith(VALID_EXTENSIONS))
        )
    )


async def _extract_audio_file_from_request(
    request: Request,
    audio: UploadFile | None,
) -> UploadFile:
    """Extract and validate audio file from request."""
    # First try the standard 'audio' parameter
    if audio is not None:
        return audio

    # iOS Shortcuts may use a different field name, scan form for audio files
    LOGGER.info("No 'audio' parameter found, scanning form fields for audio files")
    form_data = await request.form()

    for key, value in form_data.items():
        if _is_valid_audio_file(value):
            LOGGER.info("Found audio file in field '%s': %s", key, value.filename)
            return value

    # No audio file found anywhere
    raise HTTPException(
        status_code=422,
        detail="No audio file provided. Ensure the form field is named 'audio' and type is 'File'.",
    )


def _validate_audio_file(audio: UploadFile) -> None:
    """Validate audio file and return file extension."""
    if not audio or not audio.filename:
        LOGGER.error("No filename provided in request")
        raise HTTPException(status_code=400, detail="No filename provided")

    file_ext = Path(audio.filename).suffix.lower()

    if file_ext not in VALID_EXTENSIONS:
        raise HTTPException(
            status_code=400,
            detail=f"Unsupported audio format: {file_ext}. Supported: {', '.join(VALID_EXTENSIONS)}",
        )


def _load_transcription_configs() -> tuple[
    config.ProviderSelection,
    config.WyomingASR,
    config.OpenAIASR,
    config.Ollama,
    config.OpenAILLM,
    config.GeminiLLM,
    dict[str, Any],
]:
    """Load and create all required configuration objects."""
    loaded_config = config.load_config()
    wildcard_config = loaded_config.get("defaults", {})
    command_config = loaded_config.get("transcribe", {})
    defaults = {**wildcard_config, **command_config}

    provider_cfg = config.ProviderSelection(
        asr_provider=defaults.get("asr_provider", opts.ASR_PROVIDER.default),  # type: ignore[attr-defined]
        llm_provider=defaults.get("llm_provider", opts.LLM_PROVIDER.default),  # type: ignore[attr-defined]
        tts_provider=opts.TTS_PROVIDER.default,  # type: ignore[attr-defined]
    )
    wyoming_asr_cfg = config.WyomingASR(
        asr_wyoming_ip=defaults.get("asr_wyoming_ip", opts.ASR_WYOMING_IP.default),  # type: ignore[attr-defined]
        asr_wyoming_port=defaults.get("asr_wyoming_port", opts.ASR_WYOMING_PORT.default),  # type: ignore[attr-defined]
    )
    openai_asr_cfg = config.OpenAIASR(
        asr_openai_model=defaults.get("asr_openai_model", opts.ASR_OPENAI_MODEL.default),  # type: ignore[attr-defined]
        openai_api_key=defaults.get("openai_api_key", opts.OPENAI_API_KEY.default),  # type: ignore[attr-defined,union-attr]
    )
    ollama_cfg = config.Ollama(
        llm_ollama_model=defaults.get("llm_ollama_model", opts.LLM_OLLAMA_MODEL.default),  # type: ignore[attr-defined]
        llm_ollama_host=defaults.get("llm_ollama_host", opts.LLM_OLLAMA_HOST.default),  # type: ignore[attr-defined]
    )
    openai_llm_cfg = config.OpenAILLM(
        llm_openai_model=defaults.get("llm_openai_model", opts.LLM_OPENAI_MODEL.default),  # type: ignore[attr-defined]
        openai_api_key=defaults.get("openai_api_key", opts.OPENAI_API_KEY.default),  # type: ignore[attr-defined,union-attr]
        openai_base_url=defaults.get("openai_base_url", opts.OPENAI_BASE_URL.default),  # type: ignore[attr-defined,union-attr]
    )
    gemini_llm_cfg = config.GeminiLLM(
        llm_gemini_model=defaults.get("llm_gemini_model", opts.LLM_GEMINI_MODEL.default),  # type: ignore[attr-defined]
        gemini_api_key=defaults.get("gemini_api_key", opts.GEMINI_API_KEY.default),  # type: ignore[attr-defined,union-attr]
    )

    return (
        provider_cfg,
        wyoming_asr_cfg,
        openai_asr_cfg,
        ollama_cfg,
        openai_llm_cfg,
        gemini_llm_cfg,
        defaults,
    )


def _convert_audio_for_local_asr(audio_data: bytes, filename: str) -> bytes:
    """Convert audio to Wyoming format if needed for local ASR."""
    LOGGER.info("Converting %s audio to Wyoming format", filename)
    converted_data = convert_audio_to_wyoming_format(audio_data, filename)
    LOGGER.info("Audio conversion successful")
    return converted_data


async def _process_transcript_cleanup(
    raw_transcript: str,
    cleanup: bool,
    extra_instructions: str | None,
    defaults: dict[str, Any],
    provider_cfg: config.ProviderSelection,
    ollama_cfg: config.Ollama,
    openai_llm_cfg: config.OpenAILLM,
    gemini_llm_cfg: config.GeminiLLM,
) -> str | None:
    """Process transcript cleanup with LLM if requested."""
    if not cleanup:
        return None

    instructions = AGENT_INSTRUCTIONS
    config_extra = defaults.get("extra_instructions", "")
    if config_extra:
        instructions += f"\n\n{config_extra}"
    if extra_instructions:
        instructions += f"\n\n{extra_instructions}"

    return await process_and_update_clipboard(
        system_prompt=SYSTEM_PROMPT,
        agent_instructions=instructions,
        provider_cfg=provider_cfg,
        ollama_cfg=ollama_cfg,
        openai_cfg=openai_llm_cfg,
        gemini_cfg=gemini_llm_cfg,
        logger=LOGGER,
        original_text=raw_transcript,
        instruction=INSTRUCTION,
        clipboard=False,  # Don't copy to clipboard in web service
        quiet=True,
        live=None,
    )


@app.post("/transcribe", response_model=TranscriptionResponse)
async def transcribe_audio(
    request: Request,
    form_data: Annotated[TranscriptionRequest, Depends(parse_transcription_form)],
    audio: Annotated[UploadFile | None, File()] = None,
) -> TranscriptionResponse:
    """Transcribe audio file and optionally clean up the text.

    Args:
        request: FastAPI request object
        audio: Audio file (wav, mp3, m4a, etc.)
        form_data: Form data with cleanup and extra_instructions

    Returns:
        TranscriptionResponse with raw and cleaned transcripts

    """
    # Initialize variables outside try block to ensure they exist in finally block
    raw_transcript = ""
    cleaned_transcript = None

    try:
        # Extract and validate audio file
        audio_file = await _extract_audio_file_from_request(request, audio)
        _validate_audio_file(audio_file)

        # Extract form data (Pydantic handles string->bool conversion automatically)
        cleanup = form_data.cleanup
        extra_instructions = form_data.extra_instructions

        # Load all configurations
        (
            provider_cfg,
            wyoming_asr_cfg,
            openai_asr_cfg,
            ollama_cfg,
            openai_llm_cfg,
            gemini_llm_cfg,
            defaults,
        ) = _load_transcription_configs()

        # Save uploaded file
        audio_data = await audio_file.read()

        # Convert audio to Wyoming format if using local ASR
        if provider_cfg.asr_provider == "local":
            audio_data = _convert_audio_for_local_asr(audio_data, audio_file.filename)

        # Transcribe audio using the configured provider
        raw_transcript = await _transcribe_with_provider(
            audio_data,
            provider_cfg,
            wyoming_asr_cfg,
            openai_asr_cfg,
        )

        if not raw_transcript:
            return TranscriptionResponse(
                raw_transcript="",
                success=False,
                error="No transcript generated from audio",
            )

        # Process transcript cleanup if requested
        cleaned_transcript = await _process_transcript_cleanup(
            raw_transcript,
            cleanup,
            extra_instructions,
            defaults,
            provider_cfg,
            ollama_cfg,
            openai_llm_cfg,
            gemini_llm_cfg,
        )

        # If cleanup was requested but failed, indicate partial success
        if cleanup and cleaned_transcript is None:
            return TranscriptionResponse(
                raw_transcript=raw_transcript,
                cleaned_transcript=None,
                success=True,  # Transcription succeeded even if cleanup failed
                error="Transcription successful but cleanup failed. Check LLM configuration.",
            )

        return TranscriptionResponse(
            raw_transcript=raw_transcript,
            cleaned_transcript=cleaned_transcript,
            success=True,
        )

    except HTTPException:
        # Re-raise HTTPExceptions so FastAPI handles them properly
        raise
    except Exception as e:
        LOGGER.exception("Error during transcription")
        return TranscriptionResponse(raw_transcript="", success=False, error=str(e))
    finally:
        # Log the transcription automatically (even if it failed)
        # Only log if we have something to log
        if raw_transcript or cleaned_transcript:
            try:
                transcription_logger = get_default_logger()
                transcription_logger.log_transcription(
                    raw=raw_transcript,
                    processed=cleaned_transcript,
                )
            except Exception as log_error:
                LOGGER.warning("Failed to log transcription: %s", log_error)
