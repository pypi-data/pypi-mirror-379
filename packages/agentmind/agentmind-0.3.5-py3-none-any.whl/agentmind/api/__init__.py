"""
API client module for AgentMind hosted service.
"""
from .client import APIClient, AgentMindError, AuthenticationError, RateLimitError

__all__ = [
    'APIClient',
    'AgentMindError', 
    'AuthenticationError',
    'RateLimitError'
]