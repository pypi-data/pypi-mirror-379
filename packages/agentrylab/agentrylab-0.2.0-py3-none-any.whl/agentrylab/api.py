from __future__ import annotations

"""Python API entry points for Agentry Lab.

Public helpers:
  - init(config, *, experiment_id=None, prompt=None, resume=True) -> Lab
  - run(config, *, prompt=None, experiment_id=None, rounds=None, resume=True, stream=False, on_event=None) -> tuple[Lab, LabStatus]
"""

from typing import Any, Dict, Optional, Tuple, Callable

from agentrylab.config.loader import load_config
from agentrylab.lab import Lab, LabStatus, init_lab
from agentrylab.types import ProgressInfo
from agentrylab.persistence.store import Store


def init(
    config: str | Dict[str, Any] | Any,
    *,
    experiment_id: Optional[str] = None,
    prompt: Optional[str] = None,
    user_messages: Optional[list[str] | str] = None,
    resume: bool = True,
) -> Lab:
    """Initialize a Lab from a config path/dict or Preset object.

    Args:
        config: YAML path, dict, or an already-validated Preset-like object.
        experiment_id: Optional logical run/thread id (resume-able).
        prompt: Optional objective to set on the config (used by nodes if enabled).
        resume: If True, attempt to load checkpoint for experiment_id and merge.
    """
    cfg = load_config(config) if isinstance(config, (str, dict)) else config
    if prompt:
        try:
            setattr(cfg, "objective", prompt)
        except Exception:
            # Best-effort: ignore if cfg is immutable
            pass
    lab = Lab(cfg, thread_id=experiment_id, resume=resume)
    # Seed initial user message(s) into state history for context composition
    if user_messages:
        msgs = [user_messages] if isinstance(user_messages, str) else list(user_messages)
        for m in msgs:
            try:
                lab.state.history.append({"agent_id": "user", "role": "user", "content": str(m)})
            except Exception:
                pass
    return lab


def run(
    config: str | Dict[str, Any] | Any,
    *,
    prompt: Optional[str] = None,
    experiment_id: Optional[str] = None,
    rounds: Optional[int] = None,
    resume: bool = True,
    stream: bool = False,
    on_event: Optional[Callable[[Dict[str, Any]], None]] = None,
    timeout_s: Optional[float] = None,
    stop_when: Optional[Callable[[Dict[str, Any]], bool]] = None,
    on_tick: Optional[Callable[[ProgressInfo], None]] = None,
    on_round: Optional[Callable[[ProgressInfo], None]] = None,
) -> Tuple[Lab, LabStatus]:
    """Convenience: init and run the lab once.

    Returns (lab, final_status). If stream=True and on_event is provided, calls the
    callback with each new transcript event during execution.
    """
    lab = init(config, experiment_id=experiment_id, prompt=prompt, resume=resume)
    status = lab.run(
        rounds=rounds,
        stream=stream,
        on_event=on_event,
        timeout_s=timeout_s,
        stop_when=stop_when,
        on_tick=on_tick,
        on_round=on_round,
    )
    return lab, status


def list_threads(
    config: str | Dict[str, Any] | Any,
) -> list[tuple[str, float]]:
    """Return list of (thread_id, updated_at) known to the persistence store for this config.

    Accepts the same `config` kinds as `init`: path, dict, or Presetâ€‘like object.
    """
    cfg = load_config(config) if isinstance(config, (str, dict)) else config
    store = Store(cfg)
    return store.list_threads()
