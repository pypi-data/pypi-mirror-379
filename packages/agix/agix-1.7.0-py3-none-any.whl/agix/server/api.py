from __future__ import annotations

from dataclasses import asdict
from threading import Lock
from typing import Any, Dict

from fastapi import FastAPI
from fastapi.responses import JSONResponse

from src.agix.adapters.service import ServiceAdapter
from src.agix.architecture.ameta import AMetaArchitecture, AGIModule
from src.agix.memory import GestorDeMemoria


class PerceptionModule(AGIModule):
    """Módulo de percepción que pasa la observación sin cambios."""

    def process(self, observation: Any) -> Any:
        return observation


class InferenceModule(AGIModule):
    """Módulo de inferencia mínimo."""

    def process(self, data: Any) -> Any:
        return data


class DecisionModule(AGIModule):
    """Calcula una acción numérica simple a partir de la observación."""

    def process(self, inferred_state: Any) -> int:
        if isinstance(inferred_state, (list, tuple)):
            return len(inferred_state)
        if isinstance(inferred_state, (int, float)):
            return int(inferred_state)
        return 0


class MemoryModule(AGIModule):
    """Registra experiencias en ``GestorDeMemoria`` con sincronización."""

    def __init__(self, memory: GestorDeMemoria, lock: Lock) -> None:
        self.memory = memory
        self.lock = lock

    def process(self, experience: Any) -> None:
        """Registra experiencias manejando distintos formatos de entrada."""
        with self.lock:
            if isinstance(experience, tuple) and len(experience) == 2:
                obs, action = experience
                self.memory.registrar(str(obs), str(action), "", True)
            elif isinstance(experience, dict):
                self.memory.registrar(
                    str(experience.get("entrada", "")),
                    str(experience.get("decision", "")),
                    str(experience.get("resultado", "")),
                    bool(experience.get("exito", True)),
                )
            else:  # pragma: no cover - formatos no esperados
                raise ValueError("Formato de experiencia no soportado")


memory_manager = GestorDeMemoria()
memory_lock = Lock()
architecture = AMetaArchitecture(
    P=PerceptionModule(),
    I=InferenceModule(),
    D=DecisionModule(),
    M=MemoryModule(memory_manager, memory_lock),
)

service_adapter = ServiceAdapter()

app = FastAPI(title="AGIX API")


@app.post("/infer")
def infer(payload: Dict[str, Any]):
    """Procesa una observación y devuelve la acción calculada."""
    observation = service_adapter.adapt_input(payload)
    action = architecture.cycle(observation)
    return JSONResponse(service_adapter.adapt_output(action))


@app.post("/learn")
def learn(payload: Dict[str, Any]):
    """Guarda una experiencia completa proporcionada por el usuario."""
    architecture.modules["M"].process(payload)
    return JSONResponse({"status": "ok"})


@app.get("/memory")
def get_memory():
    """Devuelve las experiencias almacenadas."""
    with memory_lock:
        data = [asdict(exp) for exp in memory_manager.experiencias]
    return JSONResponse({"experiencias": data})

