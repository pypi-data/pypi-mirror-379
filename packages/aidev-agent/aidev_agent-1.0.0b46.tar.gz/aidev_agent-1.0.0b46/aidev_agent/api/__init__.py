# -*- coding: utf-8 -*-
from .abstract_client import AbstractBKAidevResourceManager
from .bk_aidev import BKAidevApi
from .ssm_client import SSMClient, get_ssm_client, get_user_access_token_from_request
from .utils import bulk_fetch

__all__ = [
    "AbstractBKAidevResourceManager",
    "BKAidevApi",
    "bulk_fetch",
    "SSMClient",
    "get_ssm_client",
    "get_user_access_token_from_request",
]
