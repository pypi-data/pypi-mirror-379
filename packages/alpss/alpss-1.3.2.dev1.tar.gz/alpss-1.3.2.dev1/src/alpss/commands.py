import argparse

from alpss.alpss_watcher import Watcher
from alpss.alpss_main import alpss_main
import os
import json
import logging
import sys

def start_watcher():
    w = Watcher()
    w.run()

def load_json_config(config):
    """Load configuration from a JSON file or return directly if it's already a dictionary."""
    if isinstance(config, dict):
        return config  # If already a dictionary, return it

    if isinstance(config, str) and os.path.exists(config):
        with open(config, "r") as file:
            return json.load(file)  # Load JSON directly

    raise ValueError(
        "Invalid config input: Provide a dictionary or a valid JSON file path."
    )

def alpss_main_with_config(config=None):
    """
    Run ALPSS with a given JSON configuration.

    Args:
        config (str or dict, optional): JSON config file, either given as parsable argument through CLI or directly as a string, or a dictionary containing config parameters.
    """

    if config is None: 
        # If called from CLI, parse arguments
        parser = argparse.ArgumentParser(
            description="Run ALPSS using a JSON config file"
        )
        parser.add_argument(
            "config_path", type=str, help="Path to the JSON configuration file"
        )
        args = parser.parse_args()
        config = load_json_config(args.config_path)

    # Load the dictionary or YAML config
    else:
        config = load_json_config(config)

    # Run ALPSS with the loaded config
    return alpss_main(**config)

def alpss_cli():
    """
    Entry point for console_scripts.
    Always uses sys.argv so `alpss /path/to/config.json` works.
    """
    try:
        sys.exit(alpss_main_with_config())
    except Exception as e:
        print(f"[ALPSS ERROR] {e}", file=sys.stderr)
        sys.exit(1)