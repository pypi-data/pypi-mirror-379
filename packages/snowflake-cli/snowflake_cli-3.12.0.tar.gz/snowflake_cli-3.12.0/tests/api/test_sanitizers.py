# Copyright (c) 2024 Snowflake Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import pytest
from snowflake.cli.api.commands.snow_typer import SnowTyper
from snowflake.cli.api.sanitizers import sanitize_for_terminal
from typer.testing import CliRunner


@pytest.mark.parametrize(
    "text, expected",
    [
        ("'\033[0i\007'", "'\x07'"),
        ("'\033[5i\007'", "'\x07'"),
        ("'ðŸ¤¯\033[1000;b\077'", "'ðŸ¤¯?'"),
        (
            "'\033[H\007''\033]1337;ClearScrollback\077''\033[2J\007''\033[1;31m'Error, enable MaliciousPlugin in your config'\033[#F\007'",
            "'\x07''1337;ClearScrollback?''\x07'''Error, enable MaliciousPlugin in your config'\x07'",
        ),
    ],
)
def test_sanitize_for_terminal(text, expected):
    result = sanitize_for_terminal(text)
    assert result == expected


def test_snow_typer_help_sanitization(os_agnostic_snapshot):
    app = SnowTyper()

    escape_text = "'\033[0i\007'"

    @app.command(epilog=escape_text, options_metavar=escape_text)
    def func1():
        """doc 1 '\033[0i\007'"""
        return 42

    @app.command(help=escape_text, rich_help_panel=escape_text, short_help=escape_text)
    def func2():
        return escape_text

    # Escape as arg not kwarg
    @app.command("'ðŸ¤¯\033[1000;b\077'")
    def func3():
        """doc 3"""
        return 42

    @app.command(name="'\033[5i\007'")
    def func4():
        """doc 4"""
        return 42

    runner = CliRunner()
    result = runner.invoke(app, ["--help"])
    assert result.output == os_agnostic_snapshot

    result = runner.invoke(app, ["func2"])
    assert result.output == ""
