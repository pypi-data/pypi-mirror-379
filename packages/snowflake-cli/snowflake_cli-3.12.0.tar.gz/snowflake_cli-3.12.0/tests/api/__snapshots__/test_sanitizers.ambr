# serializer version: 1
# name: test_snow_typer_help_sanitization
  '''
                                                                                  
   Usage: root [OPTIONS] COMMAND [ARGS]...                                        
                                                                                  
  +- Options --------------------------------------------------------------------+
  | --install-completion            Install completion for the current shell.    |
  | --show-completion               Show completion for the current shell, to    |
  |                                 copy it or customize the installation.       |
  | --help                -h        Show this message and exit.                  |
  +------------------------------------------------------------------------------+
  +- Commands -------------------------------------------------------------------+
  | ''      doc 4                                                                |
  | 'ðŸ¤¯?'   doc 3                                                                |
  | func1   doc 1 ''                                                             |
  +------------------------------------------------------------------------------+
  +- '' -------------------------------------------------------------------------+
  | func2   ''                                                                   |
  +------------------------------------------------------------------------------+
  
  
  '''
# ---
